### DISCUSSION TITLE
Persistence Problems

### DISCUSSION THREAD

**Discussion by N0tronic** [emqx/emqx#5715] [Q&A] [answered]
Hi together,

I´ve a problem with the persistence of the directory /opt/emqx/data
I use emqx version 4.3.5. Problem is, that the directories inside the data folder are hostname specific. Recomendation in emqx documentation is to set the variables EMQX_NAME and EMQX_HOST to get every time the same directory on container startup. 

Thats not practical for me, because in my configuration emqx workers are dynamic. Compose file in the attachment.
[emqx-compose.txt](https://github.com/emqx/emqx/files/7144824/emqx-compose.txt)
So I can´t use the persistent data, because I´ve every time a new hostname on conatiner startup.

Have you any idea, how I can resolve this problem?

**Comment by zmstone** [comment, marked as answer]
Hi
You can create a network to have a determined host name
https://github.com/emqx/emqx/blob/c1ae04417965c3d087a3ae32bf1e98b5312b2e62/scripts/start-two-nodes-in-docker.sh#L13-L16

**Reply by N0tronic** [reply]
Thx for the support. I know and this works fine, if I have a compose file with static defined nodes. But in my case, it is dynamic. (Look in the deploy section from attached compose file)

**Reply by zmstone** [reply]
I’m afraid EMQ X at the current stage does not support dynamic auto scale up/down that well.

in 5.0 release, there will be `core` and `replicant` node roles. Core nodes are static (like having a cluster of `master` nodes in your setup), and replicant nodes will be stateless (like the worker nodes) so you’d not need to mount volumes for them.

**Reply by N0tronic** [reply]
Thx. This workaround gives me a solution 👍

**Comment by qzhuyan** [comment]
@N0tronic what about using `--hostname` ?
https://docs.docker.com/config/containers/container-networking/

**Reply by N0tronic** [reply]
Thx for this hint. It works so long as I´ve only one Instance on each host (node). My solution now is to define two fix named manager instances of emqx and n worker instances that becomes a dynamic name. Not nice but it works.
